### PR TITLE
Move KafkaWrapper to gobblin-kafka-common

### PR DESCRIPTION
**Pull request by pcadabam**
### Changes
1. Deprecated `KafkaWrapper` in favor of `GobblinKafkaConsumerClient`. 
2. Introduced a new interface called `GobblinKafkaConsumerClient` which is wrapper on top of kafka-consumer. The interface is mostly ported from `KafkaWrapper` except that all request and response objects are defined in gobblin. `KafkaWrapper` used response objects from kafka-08 client library.
3. Added `Kafka08ConsumerClient` which is mostly code that already existed in `KafkaWrapper.KafkaOldAPI`. I have deprecated `KafkaWrapper.KafkaOldAPI` and copied its code into `Kafka08ConsumerClient`
4. Added `Kafka09ConsumerClient`
5. Added a few `ConsumerRecord` interfaces that wrap a record consumed from Kafka. `ByteArrayBasedKafkaRecord` are used by kafka-08 clients that do not support serialization and deserialization of messages. `DecodeableKafkaRecord` are used by kafka-09/10 which support serialization and deserialization.
6. Some refactor in `KafkaExtractor` to use these new `ConsumerRecords`
7. Moved a many extractors from gobblin-kafka-08 to gobblin-kafka-common

@chavdar can you review?

### CODE REVIEW

File: gobblin-modules/gobblin-kafka-08/src/main/java/gobblin/kafka/client/Kafka08ConsumerClient.java [hunk 1]
```diff
@@ -0,0 +1,390 @@
+/*
+ * Copyright (C) 2014-2016 LinkedIn Corp. All rights reserved.
```
Comment (abti) [inline gobblin-modules/gobblin-kafka-08/src/main/java/gobblin/kafka/client/Kafka08ConsumerClient.java]: Please change license header to Apache 2.0 without copyright

**Comment by pcadabam** [comment]
@abti Fixed license

Checked the failing tests. I see
```
gobblin.metastore.DatabaseJobHistoryStoreV100Test.setUp STARTED

gobblin.metastore.DatabaseJobHistoryStoreV100Test.setUp FAILED

    java.lang.RuntimeException

        Caused by: org.flywaydb.core.internal.dbsupport.FlywaySqlScriptException

            Caused by: com.mysql.jdbc.exceptions.jdbc4.MySQLSyntaxErrorException
```

File: gobblin-modules/gobblin-kafka-common/src/main/java/gobblin/kafka/client/AbstractBaseKafkaConsumerClient.java [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+import com.typesafe.config.Config;
+
+import gobblin.configuration.ConfigurationKeys;
+import gobblin.util.ConfigUtils;
+
+
+/**
+ * A base {@link GobblinKafkaConsumerClient} that sets configurations shared by all {@link GobblinKafkaConsumerClient}s
+ */
+public abstract class AbstractBaseKafkaConsumerClient implements GobblinKafkaConsumerClient {
```
Comment (chavdar) [inline gobblin-modules/gobblin-kafka-common/src/main/java/gobblin/kafka/client/AbstractBaseKafkaConsumerClient.java]: getFilteredTopics() should be movable to this class. Same applies for some of the metadata handling. The actual read of the topics metadata should be implementation-specific but there is additional post-processing that should be shareable.

Comment (pcadabam) [inline gobblin-modules/gobblin-kafka-common/src/main/java/gobblin/kafka/client/AbstractBaseKafkaConsumerClient.java]: Fixed

**Pull request merged**